MotionMark/tests/resources/main.js

```diff
@@ -223,28 +226,32 @@ Controller = Utilities.createClass(
 
     _processControllerSamples: function()
     {
+        const processedSampleFrameLengthIndex = 2;
+        const processedSampleSmoothedFrameLengthIndex = 3;
+
         var controllerSamples = new SampleData;
-        controllerSamples.addField(Strings.json.time, 0);
-        controllerSamples.addField(Strings.json.complexity, 1);
-        controllerSamples.addField(Strings.json.frameLength, 2);
-        controllerSamples.addField(Strings.json.smoothedFrameLength, 3);
+        controllerSamples.addField(Strings.json.time, sampleTimeIndex);
+        controllerSamples.addField(Strings.json.complexity, sampleComplexityIndex);
+
+        controllerSamples.addField(Strings.json.frameLength, processedSampleFrameLengthIndex);
+        controllerSamples.addField(Strings.json.smoothedFrameLength, processedSampleSmoothedFrameLengthIndex);
 
         var samples = this._sampler.samples;
-        samples[0].forEach(function(timestamp, i) {
+        samples[sampleTimeIndex].forEach(function(timestamp, i) {
             var sample = controllerSamples.createDatum();
             controllerSamples.push(sample);
 
             // Represent time in milliseconds
             controllerSamples.setFieldInDatum(sample, Strings.json.time, timestamp - this._startTimestamp);
-            controllerSamples.setFieldInDatum(sample, Strings.json.complexity, samples[1][i]);
+            controllerSamples.setFieldInDatum(sample, Strings.json.complexity, samples[sampleComplexityIndex][i]);
 
             if (i == 0)
                 controllerSamples.setFieldInDatum(sample, Strings.json.frameLength, 1000/this._targetFrameRate);
             else
-                controllerSamples.setFieldInDatum(sample, Strings.json.frameLength, timestamp - samples[0][i - 1]);
+                controllerSamples.setFieldInDatum(sample, Strings.json.frameLength, timestamp - samples[sampleTimeIndex][i - 1]);
 
-            if (samples[2][i] != -1)
-                controllerSamples.setFieldInDatum(sample, Strings.json.smoothedFrameLength, samples[2][i]);
+            if (samples[processedSampleFrameLengthIndex][i] != -1)
+                controllerSamples.setFieldInDatum(sample, Strings.json.smoothedFrameLength, samples[processedSampleFrameLengthIndex][i]);
```

Comment:
I think this should reference `samples[sampleFrameLengthEstimate][i]` since `samples` are the arrays of the `Sampler`.